### PR TITLE
Changes made for YAML input - GSoC18

### PR DESCRIPTION
**Pull request by MaidouPP**
This PR includes necessary changes for accepting YAML annotator input, so that users can provide pure YAML annotators, and RoboSherlock will take care of the conversion to xml and so on. 

The temporary xml file will generate under .ros/generated_xmls folder. The example YAML files are in annotators/io/yaml/CollectionReader.yaml and annotators/segmentation/yaml/PointCloudClusterExtractor.yaml. You can test it with u_test AE, which includes both of the CollectionReader and PointCloudClusterExtractor.

### CODE REVIEW

**Comment by bbferka** [comment]
@MaidouPP before I start reviewing please re-open the PR to the devel branch and not master;

File: src/utils/src/YamlToXMLConverter.cpp [hunk 1]
```diff
@@ -289,7 +291,9 @@ void YamlToXMLConverter::getOutput(ofstream& out) {
     out << configParams << endl;
     out << configParamSettings << endl;
     out << capabilities << endl;
-    out << "<typeSystemDescription>\n<imports>\n<import location=\"../../typesystem/all_types.xml\"/>\n</imports>\n</typeSystemDescription>\n" << endl;
+
+    string typePath = ros::package::getPath("robosherlock") + "/descriptors/typesystem/all_types.xml";
```
[the remark this answers is not in the document]
Comment (MaidouPP) [reply src/utils/src/YamlToXMLConverter.cpp]: Can I assume a consistent file sturcture like the current one? For example, an annotator yaml should be in package/descriptors/annotators/xxx/xxxx.yaml, so what I should do is to just go back for several slash and extract package?

**Comment by bbferka** [comment]
@MaidouPP tests fail because only RSControledAE class has the yaml parsing but not the RSAnalysisEngine parent class; We need to change this; The delegat parsing xml generation should happen in RSAnalysisEngine (I am  considering merging these two classes, or at least refactoring parts of the code since it's becoming quite confusing )

**Comment by MaidouPP** [comment]
Need to switch to another base branch and re PR.

**Comment by bbferka** [comment]
@MaidouPP I merged some changes from devel; please pulland fix; left a TODO fin RSControlledAE for yoy

**Pull request closed without merging**